src/components/ProductListing/ProductListing.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import { useEffect } from "react";
+
+import { useProduct } from '../../context/productcontext';
+import axios from 'axios';
+
+import "./ProductListing.css"
+export const ProductListing = () => {
+    const { product,
+        setProduct, sortBy, category, rating, range } = useProduct();
+
+    useEffect(() => {
+        (async () => {
+
+            const response = await axios.get('/api/products');
+
+            setProduct(response.data.products
+            )
+
+        })()
+
+    }, [])
+
+    const sortByPrice = (data, sortBy) => {
+
+        if (sortBy === 'LOW_TO_HIGH') {
+
+            return [...data].sort((a, b) => a.price - b.price)
+        }
+        if (sortBy === 'HIGH_TO_LOW') {
+            return [...data].sort((a, b) => b.price - a.price)
+        }
+        return data;
+    }
+
+const filterByCategory = (product, category) => {
+// console.log(mensclothing, womensclothing)
+const {   mensclothing,
+    womensclothing} = category;
+let updateMensCategory = [], updatedWomensCategory = [];
+    if(!mensclothing && !womensclothing){
+        return product;
+    }
+    if(mensclothing){
+        updateMensCategory = product.filter((item) => item.categoryName ===  "mens")
+        
+    }
+    if(womensclothing){
+        updatedWomensCategory = product.filter((item) => item.categoryName ===  "womens")
+    }
+    return [...updateMensCategory, ...updatedWomensCategory]
+
+}
+
+const sortByRating = (product, rating) =>{
+
+        if(rating === "FOUR"){
+          return  product.filter((item => item.rating > 4))
+        }
+        if(rating === "THREE"){
+          return  product.filter((item => item.rating > 3))
+        }
+        if(rating === "TWO"){
+          return  product.filter((item => item.rating > 2))
+        }
+        if(rating === "ONE"){
+          return  product.filter((item => item.rating > 1))
+        }
+    return product;
+}
+
+function filterByPriceRange(ratedProducts, range) {
+    
+    if (range) {
+      const newData = ratedProducts.filter((item) => item.price <= range);
+      return newData;
+    }
+    return ratedProducts;
+  }
+
+const getFilterByCategory = filterByCategory(product, category);
+const getSortByPrice = sortByPrice(getFilterByCategory, sortBy);
+const getRatedCategory = sortByRating(getSortByPrice, rating)
+const products = filterByPriceRange(getRatedCategory, range);
+// console.log(products)
+    return (
+        <div>
+
+            {products && products.map((item) =>
+                <div key={item._id}>
+                    <div className="product-section">
+                        <div className=" card card__vertical">
+                            <div className="image-container-vert">
+                                <img className="img responsive-image"
+                                    src={item.image}
+                                    alt="hoodie" />
```

Comment:
every image will have its own alt . so you can add that in ur db and access it here as {item.alt}